lib/receptacle/method_delegation.rb

```diff
@@ -53,86 +50,82 @@ def __build_method_call_cache(method_name)
     # @param method_cache [MethodCache] method_cache of the method to be build
     # @return [void]
     def __define_shortcut_method(method_cache)
-      define_singleton_method(method_cache.method_name) do |*args, &inner_block|
-        method_cache.strategy.new.public_send(method_cache.method_name, *args, &inner_block)
-      end
-    end
-
-    # build method to mediate method calls of arity 1 to strategy with full wrapper support
-    # @param method_cache [MethodCache] method_cache of the method to be build
-    # @return [void]
-    def __define_full_method(method_cache)
-      define_singleton_method(method_cache.method_name) do |*args, &inner_block|
-        __run_wrappers(method_cache, *args) do |*call_args|
-          method_cache.strategy.new.public_send(method_cache.method_name, *call_args, &inner_block)
+      define_singleton_method(method_cache.method_name) do |*args, **kwargs, &inner_block|
+        if kwargs.empty?
```

Comment:
For ruby versions >= 2.7 we don't need to have the two cases for every call but I added it for backward compatibility.